### ISSUE TITLE
Need wrapper to adapt tabs for mobile devices

### ISSUE THREAD

**Issue by belyan** [vmware/clarity#1590]
```
[ ] bug
[ ] feature request
[x] enhancement
```

### Expected behavior
When tabs do not fit on the screen of the mobile device, they have to hide outside of the view area of the screen. In this case the user should be able to scroll tabs forward and back via slide move.
Here is example of this behavior:
https://codepen.io/belyan/pen/gGeWEg

To realize this approach is need to add extra element around list of tabs:
```(html)
<div class="nav-wrapper">
    <ul class="nav" role="tablist">...</nav>
</div>
```

### Actual behavior
Now the tabs are positioned outside of the screen and created a horizontal scrolling.

### Reproduction of behavior
https://plnkr.co/edit/IdTbXC4P2FXgf2N1KPQN?p=preview

### Environment details

* **Angular version:** 4.3.3

* **Clarity version:** 0.10.7

* **OS and version:** iOS 11.0.3

* **Browser:** iOS 11 Safari

**Comment by belyan** [comment]
I know about the ability to hide tabs with Tab Overflow, but it is not useful for me, because tabs in my application are nicely fit on large screen. So I don't want to hide it always, I want to hide it **only in case** if it no longer fit on screen.

**Comment by adityarb88** [comment]
@belyan: But you could listen to the screen resize event and depending on the screensize, enable the tab overflow, no?

**Comment by belyan** [comment]
Yes, but it's more complex solution. Because every time when resize event is fire I have to calculate width of tabs and compare it with width of device and then find tabs which don't fit on screen and set attribute `clrTabLinkInOverflow` for it.

Also it is not good UX solution, because the user must to tap twice every time when he wants to switch between hidden tabs. It is much more convenient and familiar for the user to slide right once and when switching between tabs with one tap.

**Comment by adityarb88** [comment]
@belyan: I would argue otherwise regarding the UX feedback that you posted above. With the horizontal overflow, the user has no way to know if additional actions are present in the tab if the scrollbars are hidden. Tab Overflow clearly indicates that there is a dropdown which can be opened for additional actions.

cc: @reddolan

**Comment by adityarb88** [comment]
Also, how many times do you expect for the resize event to be fired? I wouldn't expect it to be fired a lot of times. Most of the times, if the user opens a web page on one device, the screen is not resized. Again the horizontal scroll solution can be easily created at your end but I don't believe our UX team recommends this. I will let @reddolan add his feedback to this post.

**Comment by belyan** [comment]
> With the horizontal overflow, the user has no way to know if additional actions are present in the tab if the scrollbars are hidden.

It is a common misconception. If the user sees that the content is clipped, he realizes that it is not ended and he can scrolls through further. For this reason vertical scrolling is hidden by default and appear only in the moment of interaction with the page in many mobile applications and even in a desktop browser Safari.

However, if we doubt about skills of the user we can hint him. For example, we can add the arrows:
https://material.angular.io/components/tabs/examples

**Comment by adityarb88** [comment]
@belyan: I am not sure if we can always assume that the content will be clipped. What if the scroll happens right on the whitespace? I dont believe this is a reliable solution which will work 100% of the time. Also, the codepen that you provided above hides scrollbars. Only few browsers provide that support. Firefox for eg. recently deprecated the ability to hide scrollbars. I believe that was for accessibility reasons.

Regarding the material arrow idea, @reddolan can provide more feedback.

**Comment by gnomeontherun** [comment]
You might be best served to use media queries to show/hide two different implementations of the tabs for mobile or larger screens. That way you can cater the experience, and avoid edge cases.

**Comment by belyan** [comment]
> What if the scroll happens right on the whitespace?
We can add semi-transparent background, which width will be more than width of the space between tabs. In this case the label of the corner tab will be disappearing smoothly.

> Only few browsers provide that support. Firefox for eg. recently deprecated the ability to hide scrollbars. 
You are right, but there are ways to hide scrollbar in all browsers, e.g.
https://stackoverflow.com/questions/16670931/hide-scroll-bar-but-while-still-being-able-to-scroll

**Comment by adityarb88** [comment]
@belyan: We are investigating custom scroll bars at the moment but i don't believe thats the right way to go. Custom scrollbars are a pain to get to work like native scrollbars. You have to listen to container resizes (which is not supported natively in most browsers) & screen resizes and add scroll events on overflow containers to make sure that the scrollbar is in sync with the native hidden scrollbar. Adding so much overhead to support something that the browsers dont let you customize is my opinion not a good way to go. It adds unnecessary overhead to our users applications.

**Comment by belyan** [comment]
@gnomeontherun 
You are right, I already thought about this solution. But I more like the solution when the tabs can be slide back and forward. This solution will also allow to add an animation of switching tabs in the future. Users usually like such things. :-)

**Comment by belyan** [comment]
@adityarb88 
Regarding to customize scrollbar I completely agree with you. I spent a lot of time looking for a good plugin that hides the native scrollbar and breaks nothing. I recommend to take a look at this plugin:
https://github.com/utatti/perfect-scrollbar

**Comment by adityarb88** [comment]
@belyan: Yep :-) We will be posting more updates on the Scrollbar issue soon https://github.com/vmware/clarity/issues/1522. Please subscribe to https://github.com/vmware/clarity/issues/1522 for more updates.

Our UX team will respond to your our queries and feedback here. Thanks!

**Comment by belyan** [comment]
Thank you.

**Comment by belyan** [comment]
Could you please reopen this issue, because it has not received clarification still.

**Comment by adityarb88** [comment]
cc: @reddolan
